File: app/src/main/java/se/hactar/movieregister/model/Movie.java
package se.hactar.movieregister.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

import se.hactar.movieregister.helper.imdb.ImdbHelper;

@Entity(tableName = "movie")
public class Movie {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    private int id;

    @ColumnInfo(name = "container")
    private String container;

    @ColumnInfo(name = "index")
    private String index;

    @ColumnInfo(name = "imdb_id")
    private String imdbId;

    @ColumnInfo(name = "first_name")
    private String type;

    @ColumnInfo(name = "type")
    private String name;

    @ColumnInfo(name = "year")
    private String year;

    @ColumnInfo(name = "poster_url")
    private String posterUrl;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getContainer() {
        return container;
    }

    public void setContainer(final String container) {
        this.container = container;
    }

    public String getIndex() {
        return index;
    }

    public void setIndex(final String index) {
        this.index = index;
    }

    public String getImdbId() {
        return imdbId;
    }

    public void setImdbId(final String imdbId) {
        this.imdbId = imdbId;
    }

    public String getType() {
        return type;
    }

    public void setType(final String type) {
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(final String name) {
        this.name = name;
    }

    public String getYear() {
        return year;
    }

    public void setYear(final String year) {
        this.year = year;
    }

    public String getPosterUrl() {
        return posterUrl;
    }

    public void setPosterUrl(final String posterUrl) {
        this.posterUrl = posterUrl;
    }

    @Override
    public String toString() {
        return "Movie{" +
                "container='" + container + '\'' +
                ", index='" + index + '\'' +
                ", imdbId='" + imdbId + '\'' +
                ", type='" + type + '\'' +
                ", name='" + name + '\'' +
                ", year='" + year + '\'' +
                ", posterUrl='" + posterUrl + '\'' +
                '}';
    }

    public String getImdbUrl() {
        return ImdbHelper.TITLE_URL + imdbId + "/";
    }
}

File: app/src/main/java/se/hactar/movieregister/model/Movie.kt
package se.hactar.movieregister.model

import android.util.Log
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "movie")
class Movie {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var id: Int = 0

    @ColumnInfo(name = "container")
    var container: String? = null

    @ColumnInfo(name = "index")
    var index: String? = null

    @ColumnInfo(name = "imdb_id")
    var imdbId: String? = null

    @ColumnInfo(name = "disc_type")
    var discType: String? = null

    @ColumnInfo(name = "title")
    var title: String? = null

    @ColumnInfo(name = "year")
    var year: String? = null

    @ColumnInfo(name = "index_title")
    var indexTitle: String? = null

    @ColumnInfo(name = "poster_url")
    var posterUrl: String? = null

    override fun toString(): String {
        return "Movie(id=$id, container=$container, index=$index, imdbId=$imdbId, " +
                "discType=$discType, title=$title, year=$year, indexTitle=$indexTitle, " +
                "posterUrl=$posterUrl)"
    }

    companion object {
        private val TAG = Movie::class.java.simpleName

        fun from(line: String): Movie? {
            // Example: A|1|tt0107048|DVD|Groundhog Day|1993|
            // Example: A|12|tt0080274|DVD|Shogun |1980|Disc 1
            if (!line.matches(".+\\|.+\\|.+\\|.+\\|.+\\|.+\\|.*".toRegex())) {
                Log.i(TAG, "Ignoring line that does not match import pattern, line=$line")
                return null
            }
            val columns = line.split("\\|".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            val movie = Movie().apply {
                container = get(columns, Columns.CONTAINER)
                index = get(columns, Columns.INDEX)
                imdbId = get(columns, Columns.IMDB_ID)
                discType = get(columns, Columns.DISC_TYPE)
                title = get(columns, Columns.TITLE)
                year = get(columns, Columns.YEAR)
                indexTitle = get(columns, Columns.INDEX_TITLE)
            }
            return movie
        }

        private operator fun get(columns: Array<String>, column: Columns): String {
            return if (column.ordinal < columns.size) columns[column.ordinal] else ""
        }

        private enum class Columns {
            CONTAINER, INDEX, IMDB_ID, DISC_TYPE, TITLE, YEAR, INDEX_TITLE
        }
    }
}
